batch/aiml-workloads/src/model_training.py

```diff
@@ -0,0 +1,181 @@
+        """
+        dataset_basename = Path(self._train_dataset_path).resolve().stem
+        filename = "model_cpt_{}.pkl".format(dataset_basename)
+        return filename
+
+    def _save_model(self, checkpoint_dir):
+        """
+        Accepts a directory path and returns the filepath where
+        the checkpoint for the model is saved.
+        Creates the destination directory if it does not exist.
+        """
+        # Check whether the specified path exists or not
+        isExist = os.path.exists(checkpoint_dir)
+
+        if not isExist:
+            # Create a new directory because it does not exist
+            os.makedirs(checkpoint_dir)
+
+        filename = self._get_checkpoint_name()
+        path = checkpoint_dir + filename
+
+        # Serialize the model checkpoint in to a Python Pickle file
+        with open(path, 'wb') as f:
+            pickle.dump(self._model, f)
+        return path
+
+    def train_and_save(self, checkpoint_dir):
+        """
+        Accepts a directory path.
+        Returns the filepath where the model checkpoint is saved after being
+        partially trained.
+        """
+        dataset = self._read_dataset(self._train_dataset_path)
+        features, labels = self.get_features_and_labels(dataset)
+        self._model.partial_fit(features, labels, classes=self._classes)
+        checkpoint_path = self._save_model(checkpoint_dir)
+        return checkpoint_path
+
+
+    def generate_report(self, output_path):
+        """
+        Accepts a output filepath.
+        Trains the model and appends the training report to the provided file.
+        """
+        generated_on = str(datetime.datetime.now())
+        checkpoint_name = self._get_checkpoint_name()
+        dataset_name = Path(self._train_dataset_path).resolve().name
+        train_features, train_labels = self.get_features_and_labels(
+            self._read_dataset(self._train_dataset_path)
+        )
+        test_features, test_lables = self.get_features_and_labels(
+            self._read_dataset(self._test_dataset_path)
+        )
+        training_accuracy = self.get_model_accuracy(
+            train_features,
+            train_labels
+        )
+        test_accuracy = self.get_model_accuracy(
+            test_features,
+            test_lables,
+        )
+        with open(output_path, 'a') as f:
+            report = (
+                "*****************************************************\n"
+                "Report generated on: {}\n"
+                "Training dataset: {}\n"
+                "Model checkpoint: {}\n"
+                "---\n"
+                "Accuracy on training data: {}\n"
+                "Accuracy on testing data: {}\n"
+                "\n"
+            ).format(
+                generated_on,
+                dataset_name,
+                checkpoint_name,
+                training_accuracy,
+                test_accuracy,
+            )
+            f.writelines(report)
+
+
+
+
+
+
+
+
+
+
```

Comment:
Are these stray lines intended?

Reply:
Nope, cleaned those up. Thanks for the catch!